addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/KeypadHandler.java

```diff
@@ -0,0 +1,240 @@
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, Command command) {
+        final Integer component;
+
+        switch (channelUID.getId()) {
+            case CHANNEL_BUTTON1:
+                component = COMPONENT_BUTTON1;
+                break;
+
+            case CHANNEL_BUTTON2:
+                component = COMPONENT_BUTTON2;
+                break;
+
+            case CHANNEL_BUTTON3:
+                component = COMPONENT_BUTTON3;
+                break;
+
+            case CHANNEL_BUTTON4:
+                component = COMPONENT_BUTTON4;
+                break;
+
+            case CHANNEL_BUTTON5:
+                component = COMPONENT_BUTTON5;
+                break;
+
+            case CHANNEL_BUTTON6:
+                component = COMPONENT_BUTTON6;
+                break;
+
+            case CHANNEL_BUTTON7:
+                component = COMPONENT_BUTTON7;
+                break;
+
+            case CHANNEL_BUTTONTOPRAISE:
+                component = COMPONENT_BUTTONTOPRAISE;
+                break;
+
+            case CHANNEL_BUTTONTOPLOWER:
+                component = COMPONENT_BUTTONTOPLOWER;
+                break;
+
+            case CHANNEL_BUTTONBOTTOMRAISE:
+                component = COMPONENT_BUTTONBOTTOMRAISE;
+                break;
+
+            case CHANNEL_BUTTONBOTTOMLOWER:
+                component = COMPONENT_BUTTONBOTTOMLOWER;
+                break;
+
+            case CHANNEL_LED1:
+                component = COMPONENT_LED1;
+                break;
+
+            case CHANNEL_LED2:
+                component = COMPONENT_LED2;
+                break;
+
+            case CHANNEL_LED3:
+                component = COMPONENT_LED3;
+                break;
+
+            case CHANNEL_LED4:
+                component = COMPONENT_LED4;
+                break;
+
+            case CHANNEL_LED5:
+                component = COMPONENT_LED5;
+                break;
+
+            case CHANNEL_LED6:
+                component = COMPONENT_LED6;
+                break;
+
+            case CHANNEL_LED7:
+                component = COMPONENT_LED7;
+                break;
+
+            default:
+                component = null;
+        }
+
+        if (component != null) {
+            if (isLEDComponent(component)) {
+                if (command.equals(OnOffType.ON)) {
+                    device(component, LED_STATE, LED_ON);
+                } else if (command.equals(OnOffType.OFF)) {
+                    device(component, LED_STATE, LED_OFF);
+                }
+            } else {
+                if (command.equals(OnOffType.ON)) {
+                    device(component, ACTION_PRESS);
+
+                    // Simulate button press by releasing the button. Schedule release for sometime after this method
+                    // returns.
+                    this.scheduler.schedule(new Runnable() {
```

Comment:
I guess this is a workaround for the fact that it is currently not possible to disable the autoupdate (which sets the state to ON, if it sees an ON command). If this autoupdate were disabled, this code would not be needed, right?